configure.ac

```diff
@@ -3519,6 +3520,16 @@ then
         AM_CFLAGS="-DOPENSSL_EXTRA $AM_CFLAGS"
     fi
     AM_CFLAGS="$AM_CFLAGS -DHAVE_LIGHTY -DHAVE_WOLFSSL_SSL_H=1"
+    AM_CFLAGS="$AM_CFLAGS -DHAVE_EX_DATA"
+    AM_CFLAGS="$AM_CFLAGS -DOPENSSL_ALL"
+    # recommended
+    AM_CFLAGS="$AM_CFLAGS -DOPENSSL_NO_SSL2 -DOPENSSL_NO_SSL3 -DOPENSSL_NO_COMP"
+    AM_CFLAGS="$AM_CFLAGS -DNO_OLD_TLS"
+    AM_CFLAGS="$AM_CFLAGS -DSINGLE_THREADED"
```

Comment:
@gstrauss : Why is single threaded required? This breaks use of `--enable-crlmonitor`... which should get disabled if single threaded mode.

Reply:
lighttpd is single threaded, and so lighttpd does not require the locking overhead built into wolfSSL, even though the locking overhead should be minimal compared to the crypto overhead.

Reading `configure.ac`, I see that `--enable-crl-monitor` is disabled by default, and is implicitly enabled with `--enable-jni`.  lighttpd does not enable either.

Since lighttpd does not use the CRL Monitor, would you please explain your concern about specifying -DSINGLE_THREADED for lighttpd?

Maybe, separate from this pull request, https://www.wolfssl.com/docs/wolfssl-manual/ch2/ can be updated to note that `--enable-crl-monitor` requires that `-DSINGLE_THREADED` *must not* be used.  Right now, the doc says only
> crl-monitor - enabling this option adds the ability to have wolfSSL actively monitor a specific CRL (Certificate Revocation List) directory.

Tangentially related question: does lighttpd need to take any action to re-initialize wolfSSL after lighttpd forks?  By default, lighttpd does not fork, but lighttpd can be configured to run multiple workers, which run in independent forked subprocesses after wolfSSL is configured in the main process.

Reply:
@gstrauss , I can understand lighttpd not using multi-threading. wolfSSL also does not require multi-threading. The issue is with the `./configure --enable-all`, which turns on both CRL monitor and lighttpd.

I went ahead and pushed a fix to disable CRL monitor for single threaded and not enable lighty with `--enable-all`.

I will let you know if this PR has any other Jenkins testing or peer review issues.

Thanks for your work on this!